File: spec/functional_test/fixtures/java/spring_callees/src/UserController.java
package com.test;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/users")
public class UserController {

    private final UserService service;

    public UserController(UserService service) {
        this.service = service;
    }

    @PostMapping("/")
    public String createUser() {
        String saved = service.save("hahwul");
        AuditLog.write(saved);
        return saved;
    }

    @GetMapping("/profile")
    public String profile() {
        String built = this.buildProfile();
        AuditLog.write(built);
        return built;
    }

    private String buildProfile() {
        return "profile";
    }
}

File: spec/functional_test/fixtures/kotlin/spring_callees/src/main/kotlin/com/example/UserController.kt
package com.example

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/users")
class UserController(private val service: UserService) {

    @PostMapping("/")
    fun createUser(): String {
        val saved = service.save("hahwul")
        AuditLog.write(saved)
        return saved
    }

    @GetMapping("/profile")
    fun profile(): String {
        val built = this.buildProfile()
        AuditLog.write(built)
        return built
    }

    private fun buildProfile(): String {
        return "profile"
    }
}
